src/backend/optimizer/plan/subselect.c

```diff
@@ -638,8 +638,7 @@ make_subplan(PlannerInfo *root, Query *orig_subquery, SubLinkType subLinkType,
 							&subroot,
 							config);
 
-	if ((plan->flow->locustype == CdbLocusType_SegmentGeneral ||
-		 plan->flow->locustype == CdbLocusType_General) &&
+	if ((plan->flow->locustype == CdbLocusType_General) &&
```

Comment:
I think the parentheses are redundant now.

Reply:
> the parentheses are redundant

done